include/motor-control/core/tasks/motor_hardware_task.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "motor-control/core/motor_hardware_interface.hpp"
+#include "motor-control/firmware/stepper_motor/motor_hardware.hpp"
+#include "pipettes/firmware/pipette_motor_hardware.hpp"
+
+static constexpr uint32_t TASK_STACK_DEPTH = 512;
+
+namespace motor_hardware_task {
+
+class MotorHardwareTask {
```

Comment:
These classes should be structured like the others and use `freertos_task::TaskStarter` - we purposefully invert control so that executable entry points get to choose how to build and execute the tasks so that if we need to we can use a different task framework for simulation.

Reply:
I tried to use the task starter but It would have required a big refactor because it's templated off having a message queue.

Reply:
Ah ok. Let's go with this version then I guess, although I think it would also be fine to have an unused queue.